src/Aurora.Scrapers/Option/PornhubGifsScraper.cs

```diff
@@ -1,80 +1,78 @@
-﻿namespace Aurora.Scrapers.Option;
+﻿using Aurora.Scrapers.Services;
+
+namespace Aurora.Scrapers.Option;
 
 public class PornhubGifsScraper : IOptionScraper
 {
     private const string _dateSourcePhncdn = "https://dl.phncdn.com";
 
-    private readonly IHttpClientFactory _clientProvider;
-    private readonly IOptions<ScrapersConfig> _config;
+    private readonly PagingRunner _runner;
 
-    public PornhubGifsScraper(IHttpClientFactory clientProvider, IOptions<ScrapersConfig> config)
+    public PornhubGifsScraper(PagingRunner runner)
     {
-        _clientProvider = clientProvider;
-        _config = config;
+        _runner = runner;
     }
 
     public SupportedWebsite Website => SupportedWebsite.Pornhub;
     public IEnumerable<ContentType> ContentTypes { get; init; } = new List<ContentType>() { ContentType.Gif };
 
     public async Task<List<SearchItem<SearchResultData>>> ScrapAsync(List<string> terms, CancellationToken token = default)
     {
-        var config = _config.Value;
         var baseUrl = Website.GetBaseUrl();
-        List<SearchItem<SearchResultData>> gifItems = new();
-
-        var htmlDocument = new HtmlDocument
-        {
-            OptionFixNestedTags = true
-        };
-
-        var urlsCount = 0;
-
-        using var client = _clientProvider.CreateClient(HttpClientNames.DefaultClient);
-
-        //TODO: Implement scraping of all pages
-        for (var i = 0; i < config.MaxPagesCount; i++)
-        {
-            if (config.UseLimitations && urlsCount >= config.MaxItemsCount)
+        return await _runner.RunPagingAsync(HttpClientNames.DefaultClient,
+            loadPage: async (pageNumber, client) => await TryLoadPage(terms, baseUrl, client, pageNumber),
+            scrapPage: document =>
             {
-                break;
-            }
+                var bodyNode = document.DocumentNode?.SelectSingleNode("//body");
```

Comment:
check body nullability

Reply:
We already do that on the next line of code:
 var gifLinksNodes = bodyNode?.SelectNodes("//li[contains(@class, 'gifVideoBlock')]");
 if (gifLinksNodes is not null)